File: datalib/src/main/java/nl/acidcats/tumblrlikes/data_impl/photodata/gateway_impl/greendao/filters/LeastSeenFilterOptionImpl.java
package nl.acidcats.tumblrlikes.data_impl.photodata.gateway_impl.greendao.filters;

import org.greenrobot.greendao.query.QueryBuilder;

import nl.acidcats.tumblrlikes.data_impl.photodata.gateway_impl.greendao.entities.PhotoEntity;
import nl.acidcats.tumblrlikes.db_impl_greendao.PhotoEntityDao;

/**
 * Created on 03/08/2017.
 */

public class LeastSeenFilterOptionImpl extends AbstractFilterOptionImpl {
    public LeastSeenFilterOptionImpl(PhotoEntityDao photoEntityDao) {
        setPhotoEntityDao(photoEntityDao);
    }

    @Override
    protected QueryBuilder<PhotoEntity> getQueryBuilder() {
        return _photoEntityDao.queryBuilder()
                .where(PhotoEntityDao.Properties.IsHidden.eq(false))
                .orderAsc(PhotoEntityDao.Properties.ViewCount);
    }
}

File: datalib/src/main/java/nl/acidcats/tumblrlikes/data_impl/photodata/gateway/greendao/filters/LeastSeenFilterOptionImpl.kt
package nl.acidcats.tumblrlikes.data_impl.photodata.gateway.greendao.filters

import nl.acidcats.tumblrlikes.data_impl.photodata.gateway.greendao.entities.PhotoEntity
import nl.acidcats.tumblrlikes.db_impl_greendao.PhotoEntityDao
import org.greenrobot.greendao.query.QueryBuilder

/**
 * Created on 25/10/2018.
 */
class LeastSeenFilterOptionImpl constructor(photoEntityDao: PhotoEntityDao) : AbstractFilterOptionImpl() {
    override val queryBuilder: QueryBuilder<PhotoEntity> by lazy {
        photoEntityDao.queryBuilder()
                .where(PhotoEntityDao.Properties.IsHidden.eq(false))
                .orderAsc(PhotoEntityDao.Properties.ViewCount)
    }
}
